drivers/digital-io/max22196/iio_max22196.c

```diff
@@ -0,0 +1,695 @@
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	int length = 0;
+	int i;
+
+	for (i = 0; i < NO_OS_ARRAY_SIZE(max22196_current_avail); i++)
+		length += sprintf(buf + length, "%s ", max22196_current_avail[i]);
+
+	return length;
+}
+
+/**
+ * @brief Register read wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ * @param readval - Register value
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22196_iio_reg_read(struct max22196_iio_desc *dev, uint32_t reg,
+				 uint32_t *readval)
+{
+	return max22196_reg_read(dev->max22196_desc, reg, readval);
+}
+
+/**
+ * @brief Register write wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ * @param readval - Register value
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22196_iio_reg_write(struct max22196_iio_desc *dev, uint32_t reg,
+				  uint32_t writeval)
+{
+	return max22196_reg_write(dev->max22196_desc, reg, writeval);
+}
+
+/**
+ * @brief Setup the channels for the IIO device
+ * @param desc - MAX22196 IIO device descriptor
+ * @param chans_enabled - bool array including information about each channel
+ * 			  being disabled or enabled.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+static int max22196_iio_setup_channels(struct max22196_iio_desc *desc,
+				       bool *chans_enabled)
+{
+	struct iio_channel *max22196_iio_channels;
+	uint32_t enabled_ch = 0;
+	uint32_t ch_offset = 0;
+	uint32_t i;
+
+	for (i = 0; i < MAX22196_CHANNELS; i++)
+		if (chans_enabled[i])
+			enabled_ch++;
+
+	max22196_iio_channels = no_os_calloc(enabled_ch,
+					     sizeof(*max22196_iio_channels));
+	if (!max22196_iio_channels)
+		return -ENOMEM;
+
+	for (i = 0; i < MAX22196_CHANNELS; i++)
+		if (chans_enabled[i]) {
+			max22196_iio_channels[ch_offset] = (struct iio_channel)MAX22196_CHANNEL(i);
+			max22196_iio_channels[ch_offset].attributes = max22196_attrs;
+			max22196_iio_channels[ch_offset].ch_out = 0;
+			ch_offset++;
+		}
+
+	desc->iio_dev->channels = max22196_iio_channels;
+	desc->iio_dev->num_ch = enabled_ch;
+
+	return 0;
+}
+
+/**
+ * @brief Initializes the MAX22196 IIO descriptor.
+ * @param iio_desc - The iio device descriptor.
+ * @param init_param - The structure that contains the device initial parameters.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int max22196_iio_init(struct max22196_iio_desc **iio_desc,
+		      struct max22196_iio_desc_init_param *init_param)
+{
+	struct max22196_iio_desc *descriptor;
+	int ret, i;
+
+	if (!init_param || !init_param->max22196_init_param)
+		return -EINVAL;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = max22196_init(&descriptor->max22196_desc,
+			    init_param->max22196_init_param);
+	if (ret)
+		goto free_desc;
+
+	descriptor->iio_dev = &max22196_iio_dev;
+
+	ret = max22196_iio_setup_channels(descriptor,
```

Comment:
All your channels are input. Why wouldn't you enable them all?

Reply:
Will remove chans_enabled boolean since it's not needed in this case and will setup all the channels to be enabled.